core/src/main/java/org/infinispan/security/impl/SubjectACL.java

```diff
@@ -38,4 +41,14 @@ public boolean matches(int permissionMask) {
    public String toString() {
       return "SubjectACL [roles=" + roles + ", mask=" + mask + "]";
    }
+
+   public EnumSet<AuthorizationPermission> getPermissions() {
+      EnumSet<AuthorizationPermission> permissions = EnumSet.noneOf(AuthorizationPermission.class);
+      for(AuthorizationPermission permission : AuthorizationPermission.values()) {
+         if ((mask & permission.getMask()) != 0) {
```

Comment:
this condition is incorrect. It will return `true` when `AuthorizationPermission.ALL` is checked.
```suggestion
         if (matches(permission.getMask())) {
```